stoobly_agent/app/proxy/handle_replay_service.py

```diff
@@ -66,7 +67,14 @@ def __rewrite_response(replay_context: ReplayContext):
     After replaying a request, see if the request needs to be rewritten
     """
     intercept_settings: InterceptSettings = replay_context.intercept_settings
+    flow = replay_context.flow
+    request = flow.request
+
+    has_minimized_headers = has_minimized_request_headers(request.headers)
```

Comment:
Alternate approach here is to check if request header `customer_headers.PROXY_MODE` is record. And if it is, then check record strategy